lgc/patch/FragColorExport.cpp

```diff
@@ -29,17 +29,64 @@
  ***********************************************************************************************************************
  */
 #include "lgc/patch/FragColorExport.h"
+#include "lgc/BuilderBase.h"
 #include "lgc/LgcContext.h"
+#include "lgc/patch/Patch.h"
 #include "lgc/state/IntrinsDefs.h"
+#include "lgc/state/PalMetadata.h"
+#include "lgc/state/PipelineShaders.h"
 #include "lgc/state/PipelineState.h"
 #include "lgc/state/TargetInfo.h"
 #include "llvm/IR/Constants.h"
 #include "llvm/IR/Instructions.h"
+#include "llvm/IR/IntrinsicsAMDGPU.h"
 
 #define DEBUG_TYPE "lgc-frag-color-export"
 
+using namespace lgc;
 using namespace llvm;
 
+namespace {
+
+// The information needed for an export to a hardware color target.
+struct ColorValueInfo {
+  std::vector<Value *> value; // The value of each component to be exported.
+  unsigned location;          // The location that corresponds to the hardware color target.
```

Comment:
Not sure if this could be "colorTarget" or "hwColorTarget" like what you have done.

Reply:
I'm confused once more. Originally we did these names because the `Location` in SPIR-V might not be the same as the MRT for purposes of pixel shader exports.

But now I don't see anymore where that would happen outside of dual-source blending?

Anyway, I think this is trying to be the SPIR-V `Location`, in part because `PipelineState::m_colorExportFormats` is indexed by `Location`, not by "hwColorTarget".

Reply:
You are right. I revisit the codes and find the stored `location` is from the output call. So its value is SPIR-V location. I should drop my comments. We map the SPIR-V location to MRT in the pass PatchResourceCollect. Only the value in outputLocMap[location] is the real MRT.